packages/manager-react-components/src/hooks/datagrid/useResourcesV6.tsx

```diff
@@ -60,13 +59,20 @@ function sortColumn(type: string, a: any, b: any, desc: boolean) {
   }
 }
 
-function applySearch(items, filters) {
-  if (!filters?.length) return items;
-  return items.filter((item) =>
-    filters.some(
-      ({ key, value }) =>
-        item[key]?.toString()?.toLowerCase().includes(value.toLowerCase()),
-    ),
+function applySearch(items, filters, searchInput) {
+  if (!searchInput || searchInput.length === 0 || !filters?.length)
+    return items;
+
+  return items?.filter(
+    (item) =>
+      filters?.some((columnId) => {
+        const value = item[columnId];
+        if (value === null || value === undefined) return false;
+        return value
+          ?.toString()
+          ?.toLowerCase()
+          ?.includes(searchInput?.toLowerCase());
+      }),
```

Comment:
```suggestion
  if (!searchInput?.length === 0 || !filters?.length)
    return items;
  
  const searchInputLowerCase = searchInput.toLowerCase();

  return items?.filter(
    (item) =>
      filters.some((columnId) => {
        const value = item[columnId];
        if (value === null || value === undefined) return false;
        return value
          .toString()
          .toLowerCase()
          .includes(searchInputLowerCase);
      }),
```
It avoids unused checks and avoids call to lower case for each value